src/evtx_chunk.rs

```diff
@@ -250,32 +253,36 @@ impl<'a> Iterator for IterChunkRecords<'a> {
         );
 
         let mut tokens = vec![];
-        let iter = match deserializer.iter_tokens(Some(binxml_data_size)) {
+        let iter = match deserializer.iter_tokens(Some(binxml_data_size)).context(
+            err::FailedToDeserializeRecord {
+                record_id: record_header.event_record_id,
+            },
+        ) {
             Ok(iter) => iter,
-            Err(e) => return Some(Err(format_err!("{}", e))),
+            Err(err_ctx) => return Some(Err(err_ctx.error)),
         };
 
         for token in iter {
-            match token {
+            match token.eager_context(err::FailedToDeserializeRecord {
+                record_id: record_header.event_record_id,
+            }) {
                 Ok(token) => {
                     trace!("successfully read {:?}", token);
                     tokens.push(token)
                 }
-                Err(e) => {
-                    error!("Tried to read an invalid token!, {}", e);
-
-                    if log::log_enabled!(Level::Debug) {
-                        let mut cursor = Cursor::new(self.chunk.data);
-                        cursor
-                            .seek(SeekFrom::Start(
-                                e.offset().expect("Err to have offset information"),
-                            ))
-                            .unwrap();
-                        dump_cursor(&cursor, 10);
-                    }
+                Err(err) => {
+                    //                    if log::log_enabled!(Level::Debug) {
```

Comment:
Delete?

Reply:
Is there a nice way to keep the nice hexdumps in your opinion?